packages/widget/src/components/common/buttons/connect-wallet.ts

```diff
@@ -62,49 +61,124 @@ export class ConnectWalletButton extends BaseComponent {
+          accounts: this.wallets.substrateWallet?.accounts ?? []
+        }
+      })
+    );
+  };
+
+  private renderDisconnectSubstrateButton(): HTMLTemplateResult | undefined {
+    return html` <div
+      class="dropdownOption substrateDisconnectButton"
+      part="substrateDisconnectButton"
+      @click=${this.onDisconnectClicked}
+    >
+      Disconnect
+    </div>`;
+  }
+
+  private renderSubstrateAccount(): HTMLTemplateResult {
     const substrateWallet = this.wallets.substrateWallet;
-    //TODO: this is wrong we need to enable user to select account
+    if (!substrateWallet) return html``;
+
+    function renderCustomOptionContent({
+      name,
+      address
+    }: Account): HTMLTemplateResult {
+      return html`
+        <div part="customOptionContent">
+          <div part="customOptionContentHeader">
+            <span part="customOptionContentName">${name}</span>
+          </div>
+          <div part="customOptionContentMain">
+            ${identicon}
+            <div part="customOptionContentAccountData">
+              <span part="customOptionContentAddress">${address}</span>
+            </div>
+          </div>
+        </div>
+      `;
+    }
+
+    function normalizeOptionsData(): DropdownOption<Account>[] {
+      return substrateWallet!.accounts.map((account: Account) => ({
+        id: account.address,
+        name: shortAddress(account?.address ?? ''),
+        value: account,
+        icon: greenCircleIcon,
+        customOptionHtml: renderCustomOptionContent(account)
+      }));
+    }
+
     const substrateAccount = substrateWallet?.accounts[0];
+    const options = normalizeOptionsData();
+
+    return when(
+      !!substrateAccount,
+      () =>
+        html`<dropdown-component
+          .actionOption=${this.renderDisconnectSubstrateButton()}
+          .options=${options}
+          .selectedOption=${options[0]}
+          .onOptionSelected=${this.handleSubstrateAccountSelected}
+        >
+        </dropdown-component>`
+    );
+  }
+
+  private renderConnectWalletButton(): HTMLTemplateResult | undefined {
+    if (this.isWalletConnected() && this.wallets.substrateWallet)
+      return undefined;
+
+    return when(
+      this.isWalletConnected(),
+      () =>
+        html` <button
+          role="button"
+          @click=${this.onDisconnectClicked}
+          class="connectWalletButton"
+        >
+          Disconnect
+        </button>`,
+      () =>
+        html` <button
+          @click=${this.onConnectClicked}
+          role="button"
+          class="connectWalletButton"
+        >
+          ${plusIcon} Connect Wallet
+        </button>`
+    );
+  }
+
+  private renderWalletAddress(): TemplateResult {
+    const evmWallet = this.wallets.evmWallet;
+
+    return when(
+      !!evmWallet?.address,
+      () =>
+        html`<span class="walletAddress" title=${evmWallet?.address ?? ''}
+          >${greenCircleIcon} ${shortAddress(evmWallet?.address ?? '')}</span
```

Comment:
same here

Reply:
evmWallet can be undefined and shortAddress method needs a string. So it' safer to pass the empty string

Reply:
@sztok7 Technically it shouldn't but typescript doesn't respect "when" method 😅